pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -0,0 +1,293 @@
+
+	statusHandlers := []statusHandler{memberOperatorStatusHandler, hostConnectionStatusHandler}
+
+	// track components that are not ready
+	unreadyComponents := []string{}
+
+	// retrieve component statuses eg. kubefed, member deployment
+	for _, handler := range statusHandlers {
+		err := handler.handleStatus(reqLogger, memberStatus)
+		if err != nil {
+			reqLogger.Error(err, "status update problem")
+			unreadyComponents = append(unreadyComponents, string(handler.Name))
+		}
+	}
+
+	// if any components were not ready then set the overall status to not ready
+	if len(unreadyComponents) > 0 {
+		err := fmt.Errorf("Components not ready: %v", unreadyComponents)
+		return r.setStatusNotReady(memberStatus, err.Error())
+	}
+	return r.setStatusReady(memberStatus)
+}
+
+// hostConnectionHandleStatus retrieves the host cluster object that represents the connection between this member cluster and the host cluster.
+// It then takes the status from the cluster object and adds it to MemberStatus. Finally, it checks its status and will return an error if
+// its status is not ready
+func (r *ReconcileMemberStatus) hostConnectionHandleStatus(reqLogger logr.Logger, memberStatus *toolchainv1alpha1.MemberStatus) error {
+	// look up host connection status
+	fedCluster, ok := r.getHostCluster()
+	if !ok {
+		notFoundMsg := "the host connection was not found"
+		notFoundReason := "KubefedNotFound"
+		memberStatus.Status.HostConnection = kubefed_v1beta1.KubeFedClusterStatus{
+			Conditions: []kubefed_v1beta1.ClusterCondition{
+				{
+					Type:          "Ready",
+					Status:        corev1.ConditionFalse,
+					Reason:        &notFoundReason,
+					Message:       &notFoundMsg,
+					LastProbeTime: metav1.Now(),
+				},
+			},
+		}
+		return fmt.Errorf(notFoundMsg)
+	}
+	memberStatus.Status.HostConnection = *fedCluster.ClusterStatus.DeepCopy()
+
+	// check conditions of host connection
+	if !util.IsClusterReady(fedCluster.ClusterStatus) {
+		return fmt.Errorf("the host connection is not ready")
+	}
+
+	return nil
+}
+
+// memberOperatorHandleStatus retrieves the Deployment for the member operator and adds its status to MemberStatus. It returns an error
+// if any of the conditions have a status that is not 'true'
+func (r *ReconcileMemberStatus) memberOperatorHandleStatus(reqLogger logr.Logger, memberStatus *toolchainv1alpha1.MemberStatus) error {
+	operatorStatus := toolchainv1alpha1.MemberOperatorStatus{
+		Version:        version.Version,
+		Revision:       version.Commit,
+		BuildTimestamp: version.BuildTime,
+	}
+
+	// look up status of member deployment
+	cannotGetDeploymentMsg := "unable to get the member operator deployment"
+	noDeploymentReason := "DeploymentNotFound"
+	memberOperatorDeploymentName, err := getMemberOperatorDeploymentName()
+	if err != nil {
+		errCondition := newComponentErrorCondition(noDeploymentReason, cannotGetDeploymentMsg)
```

Comment:
You lost the original `err` here. Why we can't get the deployment? Only `err` can tell you: "unable to look up the member operator name, the environment variable OPERATOR_NAME is not set".
So, I would rather do the following:
```suggestion
		errCondition := newComponentErrorCondition(noDeploymentReason, err.Error())
```
Or:
```suggestion
		err = errs.Wrap(err, cannotGetDeploymentMsg)
		errCondition := newComponentErrorCondition(noDeploymentReason, err.Error())
```

Reply:
Sorry, I meant to wrap the error. Stupid mistake :disappointed:

Fixed in ac86805b75b182c6030b9c827020aa3a58cb8624